### PR TITLE
fix(headless): fix no results back analytics event

### PR DESCRIPTION
**Pull request by olamothe**
https://coveord.atlassian.net/browse/KIT-904

### CODE REVIEW

File: packages/headless/src/controllers/history-manager/headless-history-manager.test.ts [hunk 1]
```diff
@@ -52,6 +57,17 @@ describe('History Manager', () => {
...
+  it('should allow to navigate backward on no results when there is a past history', () => {
+    engine.state.history.present = getHistoryInitialState();
+    engine.state.history.past = [extractHistory({pipeline: 'test'})];
+    initHistoryManager();
+
+    historyManager.backOnNoResults();
+
+    expect(engine.actions[0].type).toBe(back.pending.type);
+    expect(engine.actions[1].type).toBe(undo().type);
+  });
```
Comment (samisayegh) [inline packages/headless/src/controllers/history-manager/headless-history-manager.test.ts:69]: Where does the `undo` action come from? I do not see it dispatched [here](https://github.com/coveo/ui-kit/pull/1202/files#diff-8b720ca80c59926649cfac18e286e82eb394d213a27c3bdc8ba6eb0b0a1e1169R95).

Also, should there be an assertion for the `executeSearch` action?

**Review by samisayegh — APPROVED** [review]

File: packages/headless/src/controllers/history-manager/headless-history-manager.test.ts (hunk 1 above)
Comment (olamothe) [reply packages/headless/src/controllers/history-manager/headless-history-manager.test.ts:69]: `back()` is dispatching two action under the hood: `undo` (move back in history) + `change` (update all slice to the new present state).

I can add a check on `executeSearch` also

File: packages/atomic/src/components/atomic-no-results/atomic-no-results.tsx [hunk 2]
```diff
@@ -102,7 +102,7 @@ export class AtomicNoResults {
         style="primary"
         part="cancel-button"
         text={this.bindings.i18n.t('cancel-last-action')}
-        onClick={() => this.history.back()}
+        onClick={() => this.history.backOnNoResults()}
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/atomic-no-results/atomic-no-results.tsx:105]: Question I had when implementing this, is the history manager controller really necessary? Or we can use the browser history to achieve this?

**Review by ThibodeauJF — APPROVED** [review]

File: packages/atomic/src/components/atomic-no-results/atomic-no-results.tsx (hunk 2 above)
Comment (olamothe) [reply packages/atomic/src/components/atomic-no-results/atomic-no-results.tsx:105]: > Question I had when implementing this, is the history manager controller really necessary? Or we can use the browser history to achieve this?

The difference with the history manager is that it only creates "snapshot" for "successful" state (those that returned results and that did not return a query error).

Versus browser history, which could very well just put you in a "loop" with no results or with query errors.

**Pull request merged**